library/std/src/sys/pal/unix/thread.rs

```diff
@@ -68,7 +68,14 @@ impl Thread {
                     let page_size = os::page_size();
                     let stack_size =
                         (stack_size + page_size - 1) & (-(page_size as isize - 1) as usize - 1);
-                    assert_eq!(libc::pthread_attr_setstacksize(attr.as_mut_ptr(), stack_size), 0);
+
+                    // Some libc implementations, e.g. musl, place an upper bound
+                    // on the stack size, in which case we can only gracefully return
+                    // an error here.
+                    if libc::pthread_attr_setstacksize(attr.as_mut_ptr(), stack_size) != 0 {
+                        drop(Box::from_raw(p));
```

Comment:
Doesn't this miss a call to `pthread_attr_destroy`?

Reply:
Good catch, that was indeed missing :)